tests/unit_tests/blas/batch/axpy_batch_usm.cpp

```diff
@@ -54,7 +54,7 @@ int test(device *dev, oneapi::mkl::layout layout, int64_t group_count) {
             catch (exception const &e) {
                 std::cout << "Caught asynchronous SYCL exception during AXPY_BATCH:\n"
                           << e.what() << std::endl
-                          << "OpenCL status: " << e.get_cl_code() << std::endl;
+                          << "OpenCL status: " << e.what() << std::endl;
```

Comment:
I guess with new exception method first part of the error message `OpenCL status:` will be a little bit misleading. Should it be removed completely in this case?

Reply:
What do you think about using `Error status`? I think it would be more elegant?

Reply:
I don't think it makes sense to print e.what() twice. 
Also, I see get_cl_code is part of SYCL 1.2.1 spec (section 4.9). (Please correct me if I am missing something)
Anything you can do to support this from hipSYCL side? It can be a dummy call?
I think, providing the OpenCL error status is a valuable debug feature for DPC++

Reply:
It's part of the SYCL 1.2.1 OpenCL interoperability API which doesn't make sense in hipSYCL since hipSYCL does not have an OpenCL backend. We could add a dummy call if this is a blocker, but I'd prefer not to. 
I think the better strategy might be to switch to the SYCL 2020 exception interface in the mid term, which provides a replacement for `get_cl_code()`: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:exception.class

Reply:
I'm fine with using `Error status: e.status()`. Let me check if it works with DPC++ compiler as well.

Reply:
I've added the wrapper for the `get_cl_code`. note that sycl2020 exception handling is still missing. So we probably want to add that before this is merged.

Reply:
Thank you! Looks good to me. Could you please clarify what exact changes we need for sycl2020 exception handling?

Reply:
I am sorry for the confusion I meant we need to add the exception handling to hipSYCL, but that should be handled by this PR https://github.com/illuhad/hipSYCL/pull/595 . I don't think that there is any modification necessary from the side of oneMKL or dpc++

Reply:
Oh, I see, thank you!

Reply:
Looks good to me. Could you please rebase and add similar changes to newer tests?

Reply:
And please re-run clang-format, I've just noticed it doesn't like how the changes look after `print_error_code` function introduction.

Reply:
Hi @sbalint98, did you have a chance to re-run clang-format for this PR? Everything else look good to me for integration.

Reply:
yes, I did that, before rebasing. I also checked it again, everything seems to be well formatted.